File: 141-linked-list-cycle/linked-list-cycle.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution(object):
    def hasCycle(self, head):
        hashset=set()
        while(head!=None):
            if head.next in hashset:
                return True
            hashset.add(head)
            head=head.next
        return False
        

File: 141-linked-list-cycle/linked-list-cycle.js
/**
 * Definition for singly-linked list.
 * function ListNode(val) {
 *     this.val = val;
 *     this.next = null;
 * }
 */

/**
 * @param {ListNode} head
 * @return {boolean}
 */
var hasCycle = function(head) {
    let l=head;
    let r=head;
    while(r!=null &&r.next!=null){
        
        l=l.next;
        r=r.next.next;
        if(l==r){return true;}
    }
    return false;
    
};